### ISSUE TITLE
Can't install on Debian 12 - libc version issue

### ISSUE THREAD

**Issue by lgwapnitsky** [o2sh/onefetch#1460]
### Duplicates

- [X] I have searched the existing issues

### Current behavior 😯

Installation of the .deb file fails due to a dependency issue. Debian 12 does not currently have required version of libc

```
dpkg -i onefetch_2.22.0_amd64.deb
Selecting previously unselected package onefetch.
(Reading database ... 42987 files and directories currently installed.)
Preparing to unpack onefetch_2.22.0_amd64.deb ...
Unpacking onefetch (2.22.0) ...
dpkg: dependency problems prevent configuration of onefetch:
 onefetch depends on libc6 (>= 2.39); however:
  Version of libc6:amd64 on system is 2.36-9+deb12u8.

dpkg: error processing package onefetch (--install):
 dependency problems - leaving unconfigured
Processing triggers for man-db (2.11.2-2) ...
Errors were encountered while processing:
 onefetch
```

### Expected behavior 🤔

Onefetch should operate on the current version of libc 

### Steps to reproduce 🕹

See current behavior

**Comment by spenserblack** [comment]
Related conversation: https://github.com/o2sh/onefetch/issues/1125

Could you try installing via `cargo install onefetch` and let us know if that worked?

**Comment by lgwapnitsky** [comment]
sure.  it'll be a few days.

I'm trying to avoid cargo, as I'm playing around with ansible, and trying
to make my installs as lightweight as possible

*Larry G. Wapnitsky*


*E: ***@***.****
*Web: Larry.Wapnitsky.com <http://larry.wapnitsky.com/>*

**Comment by spenserblack** [comment]
Thanks! I actually have a PR up to use Bookworm (Debian 12) in a devcontainer: https://github.com/o2sh/onefetch/pull/1461. You could try starting up a GitHub Codespace to see if you can compile from source on Debian 12. I'm not sure which version of libc the container would have, but it's a good start to at least confirm that onefetch can be built on Debian 12. You might also try to install the `.deb` package in that container. That way you don't clutter up your main environment, and we can investigate the differences between that container and your install if it works in one but not the other.

**Comment by Aldriik0** [comment]
This issue still occurs even after pulling the new version. Running `cargo build` works, however, running `cargo install` is greeted with multiple `trait mismatch` and `trait bound` not satisfied errors

**Comment by spenserblack** [comment]
> Running `cargo build` works, however, running `cargo install` is greeted with multiple `trait mismatch` and `trait bound` not satisfied errors

:thinking: If it's buildable then it should be installable. What's the output of `cargo -V`?

**Comment by Aldriik0** [comment]
cargo 1.86.0 (adf9b6ad1 2025-02-28) is the result of `cargo -V`

**Comment by spenserblack** [comment]
What are the full commands? `cargo build` and `cargo install --path .` (installing the locally cloned version)? Or `cargo install onefetch` (installing the remote version).

Either way Rust should never have a breaking version, so whatever successfully built on a previous version should also successfully install.

Regardless, I think this is a different issue than this one, so if it persists could you open as a new issue?
